virtool/samples/api.py

```diff
@@ -707,16 +704,15 @@ async def upload_cache_reads(req):
     name = req.match_info["filename"]
     sample_id = req.match_info["sample_id"]
     key = req.match_info["key"]
+    config = get_config_from_req(req)
 
     if name not in ["reads_1.fq.gz", "reads_2.fq.gz"]:
         raise HTTPBadRequest(text="File name is not an accepted reads file")
 
-    caches_path = req.app["config"].data_path / "caches"
+    caches_path = join_cache_path(config, key)
```

Comment:
Naming is not great here.

For `caches_path`; confusing as `join_cache_path` returns the path to a single, specific cache (singular). This makes more sense as `cache_path`.